### PR TITLE
DEPR: method, limit, fill_axis keywords in align

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

xref #51856, closes if we decide we don't want to deprecate broadcast_axis

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -9355,15 +9355,23 @@ def align(
             - pad / ffill: propagate last valid observation forward to next valid.
             - backfill / bfill: use NEXT valid observation to fill gap.
 
+            .. deprecated:: 2.1
+
         limit : int, default None
```
Comment (phofl) [inline pandas/core/generic.py:9360]: Should we change the defaults to lib.no_default? (for all keywords)

Comment (jbrockmendel) [reply pandas/core/generic.py:9360]: i guess thatd be harmless, OK

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @jbrockmendel

File: pandas/core/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/generic.py:9354]: good call here, we had a few places where we explicitly passed method=None that werent caught in the previous version.

want to weigh in on broadcast_axis in #51856?

**Pull request merged**